core/clock/impl/ticker_impl.cpp

```diff
@@ -33,16 +33,20 @@ namespace kagome::clock {
     return started_;
   }
 
+  clock::SystemClock::Duration TickerImpl::interval() {
+    return interval_;
+  }
+
   void TickerImpl::asyncCallRepeatedly(
       std::function<void(const std::error_code &)> h) {
-    if(not started_) {
-      callback_ = std::move(h);
-    }
+    callback_ = std::move(h);
   }
 
   void TickerImpl::onTick(const boost::system::error_code &ec) {
     callback_(ec);
-    timer_.expires_from_now(interval_);
-    timer_.async_wait(boost::bind(&TickerImpl::onTick, this, ec));
+    if (started_) {
+      timer_.expires_after(interval_);
+      timer_.async_wait(boost::bind(&TickerImpl::onTick, this, ec));
```

Comment:
Might be better to use single way for callback. We are usually using lambdas instead boost::bind